plugins/dicom_viewer/server/event_helper.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+import threading
+
+from girder import events
+
+
+class EventHelper(object):
+    """
+    Helper class to wait for plugin's data.process event handler to complete.
+    Usage:
+        with EventHelper('event.name') as helper:
+            self.model('upload').uploadFile(...)
+            handled = helper.wait()
+    """
```

Comment:
Since this file is copied directly from my repository, can you:
1. Split out adding this class into its own commit
1. Include the following in the commit message:
```
From: Max Smolens <max.smolens@kitware.com>
```

Reply:
Of course, maybe this class should be integrate into Girder Core, it's really a great tools to manage event.
I will discuss about that with @brianhelba

Reply:
Honestly, this functionality should be upstreamed into core Girder. @msmolens if you have time to do so soon, we can just merge the new feature into this branch. Otherwise, we can move forward with a temporary implementation in this plugin  (crediting the source), and move it upstream later.

Reply:
@msmolens As much as I agree with making this functionality into its own commit in principle, it might be too difficult to do so, since this is a collaborative branch, and rewriting history would be disruptive. @Pierre-Assemat and I will see what we can do, however.